pkg/scheduler/partition.go

```diff
@@ -1286,12 +1291,54 @@ func (pc *PartitionContext) UpdateAllocation(alloc *objects.Allocation) (request
 	return false, false, nil
 }
 
+func (pc *PartitionContext) handleForeignAllocation(allocationKey, applicationID, nodeID string, node *objects.Node, alloc *objects.Allocation) (requestCreated bool, allocCreated bool, err error) {
+	allocated := alloc.IsAllocated()
+	if !allocated {
+		return false, false, fmt.Errorf("trying to add a foreign request (non-allocation) %s", allocationKey)
+	}
+	if alloc.GetNodeID() == "" {
+		return false, false, fmt.Errorf("node ID is empty for allocation %s", allocationKey)
+	}
+	if node == nil {
+		return false, false, fmt.Errorf("failed to find node %s for allocation %s", nodeID, allocationKey)
+	}
```

Comment:
Correct, it's sent back and it needs to be handled properly.
Currently, this happens inside `scheduler_callback.go`:

```
	for _, reject := range response.RejectedAllocations {
		// request rejected by the scheduler, reject it
		log.Log(log.ShimRMCallback).Debug("callback: response to rejected allocation",
			zap.String("allocationKey", reject.AllocationKey))
		dispatcher.Dispatch(NewRejectTaskEvent(reject.ApplicationID, reject.AllocationKey,
			fmt.Sprintf("task %s allocation from application %s is rejected by scheduler",
				reject.AllocationKey, reject.ApplicationID)))
	}
```

This doesn't look right, we don't need a task reject event for a non-Yunikorn allocation. Therefore, we need changes in this code path (my feeling is that there's not much we can do other than logging something) but, this is yet to be done.

Reply:
Agree, there is nothing we can do except logging.

 If we want to log the information back to the Pod (Pod event? or other places?), we need to determine whether "This rejected allocation is YuniKorn or Non-YuniKorn." How do we identify it?

Will we keep the Non-YuniKorn allocation ID in shim?
My previous question assumes we don't keep the Non-YuniKorn allocation ID list in shim.

> Besides, should we add a flag of allocation type to &si.RejectedAllocation?

 Therefor, we might need a flag from si.RejectedAllocation.

Reply:
The only way we would *ever* reject a non-YK alloc in the core is if the node was not found. This would indicate a bug in the shim, and so there's no appropriate way to handle it. So logging loudly is probably sufficient.